app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -1472,7 +1476,9 @@ abstract class BaseBrowserFragment :
     override fun onConfigurationChanged(newConfig: Configuration) {
         super.onConfigurationChanged(newConfig)
 
-        _browserToolbarView?.dismissMenu()
+        _browserToolbarView?.run {
+            onUpdateToolbarForConfigurationChange(this)
+        }
```

Comment:
nit: simplify. :)

```suggestion
        _browserToolbarView?.onUpdateToolbarForConfigurationChange(this)
```

Reply:
actually this is a good callout and bigger than a nit. Your suggestion wouldn't work because this really should change to `?.let` since it was just a null check and shadowing will be fine.

Reply:
> Thanks for this comment. I was looking at this and in my head, I was thinking using `run` is a bit peculiar because I was thinking "huh this is like the same as `let`" and at the same time, I am all for simplicity.
`
Somehow I missed your comment which came first. You were correct, I think @jonalmeida was saying this should be called on the `_browserToolbarView` since he saw a `run` but this needs to be a `let` as you were thinking.

Reply:
it does resolve the null pointer issue! It works fine and would be safe to merge as-is but `run` is more commonly used to return the changed object from the calling block (as in a `run` to change fields in an object and return that object). Because it was assumed that `_browserToolbarView?.onUpdateToolbarForConfigurationChange(this)` would work it tells me at a glance people will believe this is being called on itself when in reality it is not. This is pseudo-incorrect code (works fine but not convention) so I will change it to what it should be :)

Reply:
Hmm.. reading https://kotlinlang.org/docs/scope-functions.html#run states that it invokes with `let` so I think using `run` is fine, but `let` would probably make this clearer to me.

Reply:
@jonalmeida your original suggestion wouldn't work because `onUpdateToolbarForConfigurationChange` is a function in `BaseBrowserFragment`

Reply:
@gabrielluong you're correct and this is completely safe as you mentioned. The difference is just in convention and what is returned (which in this case is not used). If we use `let` here then the result of the lambda is returned but if we use `run` like I did we return the resulting object which _could_ be useful in the future which is why I initially chose it but for readability `?.let` is maybe better.

Reply:
Oh yes, sorry - was fooled too by this call. Semantics aside between run/let, maybe we need a comment here since two of us have been caught by this call.

Architectural context: We usually try to avoid inheritance-like structures or build scaffolding (see [AC components architecture docs](https://mozac.org/contributing/architecture) and `UserInteractionHandler` for smaller code examples) around it so you we don't end up writing fixes that involve inheritance because it can be confusing. There are times when you can't fight it because Android/Java is built that way, and this might be one of those cases. Where I don't have good suggestions to void in a crash-fix PR. :)

Reply:
Let's file a followup to investigate whether or not we can avoid the inheritance-like structure.  The change is simple enough right now for the crash fix. We can look at the follow investigation afterwards.

Reply:
Agreed but if @gabrielluong agrees I want to change this to `?.let` so it's clear, readable, and doesn't require a comment to understand.

About inheritence structures I completely agree and I actually have a separate branch done without it that we can debate whether it is better or not in the future. I think it still mostly makes sense for this and considering future changes but look forward to hearing other thoughts!

Reply:
I think we can go with`?.let` without the extra comment that should be clear enough (hopefully).